adapters/adview/adview.go

```diff
@@ -157,3 +177,42 @@ func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType,
 	}
 	return mediaType, nil
 }
+
+func getMediaTypeForBid(bid openrtb2.Bid, reqImps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	mediaType := openrtb_ext.BidTypeBanner
+	for _, imp := range reqImps {
+		if bid.ImpID == imp.ID {
+			if bid.Ext != nil {
+				var bidExt *adviewBidExt
+				err := json.Unmarshal(bid.Ext, &bidExt)
+				if err == nil {
+					return getBidType(bidExt), nil
+				} else {
+					return mediaType, err
+				}
+			}
+			return mediaType, nil
+		}
+	}
+	return mediaType, nil
+}
+
+// getBidType returns the bid type specified in the response bid.ext
+func getBidType(bidExt *adviewBidExt) openrtb_ext.BidType {
+	// setting "banner" as the default bid type
+	bidType := openrtb_ext.BidTypeBanner
+	if bidExt != nil {
+		switch bidExt.BidType {
+		case 0:
+			bidType = openrtb_ext.BidTypeBanner
+		case 1:
+			bidType = openrtb_ext.BidTypeVideo
+		case 2:
+			bidType = openrtb_ext.BidTypeNative
+		default:
+			// default is banner
+			bidType = openrtb_ext.BidTypeBanner
```

Comment:
should error if none of the cases match

Reply:
I have confirmed that adview server side just support banner,video&native, there is no other value will be returned for response, so default case will not occur. thanks.

Reply:
Adding default case and returning error will help to identify case if adview server is not returing media type. Therefore will insist to return error if none of the switch case match

Reply:
Ok, agree with you and i committed the changes. pls check it again.